src/metabase/upload.clj

```diff
@@ -589,10 +590,59 @@
   [db table]
   (nil? (can-update-error db table)))
 
+(defn- check-can-delete
+  "Throws an error if the given table is not an upload, or if the user does not have permission to delete it."
+  [db table]
+  ;; For now anyone that can update a table is allowed to delete it.
+  (when-let [error (can-update-error db table)]
+    (throw error)))
+
 ;;; +--------------------------------------------------
 ;;; |  public interface for updating an uploaded table
 ;;; +--------------------------------------------------
 
+(def ^:private table-referencing-models
+  (remove nil?
+          [:model/Field
```

Comment:
Hmm, these models are not cascadely deleted?

Reply:
I'm not sure what you mean? I do cascade deletes for Field. I don't think any of the other models have children, but I didn't check exhaustively.

Reply:
I mean I don't think we need to delete these manually. As deleting table will automatically delete all of these child models since they're FKs with on delete cascade policy.

Reply:
Ah gotcha, we have cascading deletes on our FKs. I've bee conditioned to avoid FKs, and didn't expect them 😄